tests/purs/failing/2109-discard.out

```diff
@@ -0,0 +1,10 @@
+Error found:
+in module [33mMain[0m
+at tests/purs/failing/2109-discard.purs:7:3 - 7:12 (line 7, column 3 - line 7, column 12)
+
+  Unknown value [33mdiscard[0m. You're probably using do-notation, which the compiler replaces with calls to the [33mdiscard[0m function. Please import [33mdiscard[0m from module [33mPrelude[0m
```

Comment:
I feel like the error for `discard` is not ideal. Do-notation does not compile to calls to `discard`, but to `bind`. Should I change this in another PR?

Reply:
> Do-notation does not compile to calls to discard, but to bind. Should I change this in another PR?

The compiler compiles this expression:
```
do
  x <- act1
  act2 x
  act3
```
to `bind act1 (\x -> discard (act2 x) (\_ -> act3))`

Reply:
I know that it does. My point is that this error message implies that do-notation compiles primarily to `discard`, where the most important function for do-notation is `bind`.

Reply:
I agree, I think we can make this slightly clearer/less potentially misleading. Addressing this in another PR would be good, yeah.